File: MaximumSubarray - py/maximumSubarray.py
class Solution:
    """
    @param nums: A list of integers
    @return: A integer indicate the sum of max subarray
    """
    def maxSubArray(self, nums):
        # write your code here
        sum=0;rslt=-99999999;
        for i in range(0,len(nums)):
            sum += nums[i]
            rslt = max(rslt,sum )
            sum = max(sum,0)
        return rslt

if __name__ == '__main__':
    se =Solution()
    print(se.maxSubArray([-1,-1,-1]))

File: MaximumSubarray/MaximumSubarray.cpp

using namespace std;
#include<iostream>
#include<vector>

class Solution {
public:
	/**
	* @param nums: A list of integers
	* @return: A integer indicate the sum of max subarray
	*/
	int maxSubArray(vector<int> &nums) {
		// write your code here
		int ans = -1111111;
		int sum = 0;
		for (int i = 0; i < nums.size(); i++) {
			sum += nums[i];
			if (sum > ans) {
				ans = sum;
			}
			if (sum < 0) {
				sum = 0;
			}
		}
		return ans;
	}
};

int main() {
	return 0;
}